flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/result/CollectStreamResult.java

```diff
@@ -86,18 +86,17 @@ public CollectStreamResult(
 	}
 
 	@Override
-	public void startRetrieval(ProgramDeployer deployer) {
+	public void startRetrieval(JobClient jobClient) {
 		// start listener thread
 		retrievalThread.start();
 
-		jobExecutionResultFuture = deployer
-				.deploy()
-				.thenCompose(jobClient -> jobClient.getJobExecutionResult(classLoader))
+		jobExecutionResultFuture = CompletableFuture.completedFuture(jobClient)
```

Comment:
this should just be `jobClient.getJobExecutionResult(classLoader)`, no need to wrap the jobClient in a future and then compose sth to it.